src/core_plugins/metric_vis/public/metric_vis_controller.js

```diff
@@ -156,8 +159,8 @@ export class MetricVisComponent extends Component {
     if (!metric.filterKey || !metric.bucketAgg) {
       return;
     }
-    const filter = metric.bucketAgg.createFilter(metric.filterKey);
-    this.props.vis.API.queryFilter.addFilters(filter);
+    const table = this.props.visData.tables[metric.tableIndex];
+    this._vis.API.events.addFilter(table, metric.columnIndex, metric.rowIndex);
```

Comment:
I think that's a mistake and should still be `this.props.vis`

Reply:
Even after fixing that, filtering on metric does not work anymore at all. Neither on `Other` bucket nor an any other split bucket.